provider/blob_online.go

```diff
@@ -0,0 +1,204 @@
+	onlineBlobConfig := &OnlineBlobConfig{}
+	if err := onlineBlobConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	return NewOnlineBlobStore(onlineBlobConfig)
+}
+
+func NewOnlineBlobStore(config *OnlineBlobConfig) (*OnlineBlobStore, error) {
+	blobStore, err := CreateBlobStore(string(config.Type), Config(config.Config))
+	if err != nil {
+		return nil, fmt.Errorf("could not create blob store: %v", err)
+	}
+	return &OnlineBlobStore{
+		blobStore,
+		BaseProvider{
+			ProviderType:   BlobOnline,
+			ProviderConfig: config.Serialized(),
+		},
+	}, nil
+}
+
+func (store *OnlineBlobStore) AsOnlineStore() (OnlineStore, error) {
+	return store, nil
+}
+
+func blobTableKey(feature, variant string) string {
+	return fmt.Sprintf("tables/%s/%s", feature, variant)
+}
+
+func (store OnlineBlobStore) tableExists(feature, variant string) (bool, error) {
+	tableKey := blobTableKey(feature, variant)
+	return store.Exists(tableKey)
+}
+
+func (store OnlineBlobStore) readTableValue(feature, variant string) (ValueType, error) {
+	tableKey := blobTableKey(feature, variant)
+	value, err := store.Read(tableKey)
+	if err != nil {
+		return NilType, err
+	}
+	return ValueType(string(value)), nil
+}
+
+func (store OnlineBlobStore) writeTableValue(feature, variant string, valueType ValueType) error {
+	tableKey := blobTableKey(feature, variant)
+	return store.Write(tableKey, []byte(valueType))
+}
+
+func (store OnlineBlobStore) deleteTable(feature, variant string) error {
+	tableKey := blobTableKey(feature, variant)
+	return store.Delete(tableKey)
+}
+
+func (store OnlineBlobStore) GetTable(feature, variant string) (OnlineStoreTable, error) {
+	exists, err := store.tableExists(feature, variant)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, &TableNotFound{feature, variant}
+	}
+	tableType, err := store.readTableValue(feature, variant)
+	if err != nil {
+		return nil, err
+	}
+	return OnlineBlobStoreTable{store, feature, variant, tableType}, nil
+}
+
+func (store OnlineBlobStore) CreateTable(feature, variant string, valueType ValueType) (OnlineStoreTable, error) {
+	exists, err := store.tableExists(feature, variant)
+	if err != nil {
+		return nil, err
+	}
+	if exists {
+		return nil, &TableAlreadyExists{feature, variant}
+	}
+	if err := store.writeTableValue(feature, variant, valueType); err != nil {
+		return nil, err
+	}
+	return OnlineBlobStoreTable{store, feature, variant, valueType}, nil
+}
+
+type OnlineBlobStoreTable struct {
+	store     BlobStore
+	feature   string
+	variant   string
+	valueType ValueType
+}
+
+func (store OnlineBlobStore) DeleteTable(feature, variant string) error {
+	exists, err := store.tableExists(feature, variant)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return &TableNotFound{feature, variant}
+	}
+	return store.deleteTable(feature, variant)
+	//TODO should this also cycle through all values and delete them too?
```

Comment:
What values is this referring to? In the file?

Reply:
yes. The "index" for all the tables is in /tables while the actual values for the entities are in /values. I can delete the tables but the values persist. Really it makes sense to get rid of everything so I'll add a directory delete as par tof the delete table

Reply:
Yeah makes sense